### PR TITLE
Kan 3 create user repository and tests

### PR DESCRIPTION
**Pull request by Casiotone**
Creating of user repository and tests

### CODE REVIEW

**Review by nekocho — APPROVED** [review]

File: lib/user.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+        self.id = id 
+        self.username = username
+        self.email = email
+        self.password = password
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__  
+
+    def __repr__(self):
+        return f'User({self.id}, {self.username}, {self.email}), {self.password}'
```
Comment (Sabirah42) [inline lib/user.py:12]: Is this closing bracket in the right place?

File: lib/user_repository.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def all(self):
+        rows = self.connection.execute('SELECT * from users')
+        users = []
+        for row in rows:
+            item = User(row["id"], row["username"], row["email"], row['password'])
+            users.append(item)
+        return users
+
+    def create(self, user):
+        rows = self.connection.execute('INSERT INTO users (username, email, password) VALUES (%s, %s, %s) RETURNING id', [
```
Comment (Sabirah42) [inline lib/user_repository.py:17]: Ooh, I've not come across 'RETURNING id' before! Could we talk about it at the next meeting?

File: lib/user_repository.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                                 user.username, user.email, user.password])
+        return rows[0]['id']
```
Comment (Sabirah42) [inline lib/user_repository.py:19]: Probably linked to above, but why are we returning on a Create func?

**Review by Sabirah42 — COMMENTED** [review]
I've added some comments for review

**Pull request merged**